### PR TITLE
Increase delta-kernel-rust version to 0.2

### PR DESCRIPTION
**Pull request by pranavsuku-db**
Support for DV and column mapping with column partition

### CODE REVIEW

File: python/delta-kernel-python/src/lib.rs [hunk 1]
```diff
@@ -88,59 +87,38 @@ impl Scan {
         &self,
         engine_interface: &PythonInterface,
     ) -> DeltaPyResult<PyArrowType<Box<dyn RecordBatchReader + Send>>> {
-        let mut results = self.0.execute(engine_interface.0.as_ref())?;
-        match results.pop() {
-            Some(last) => {
-                // TODO(nick): This is a terrible way to have to get the schema
-                let rb: RecordBatch = last
-                    .raw_data?
+        let result_schema: SchemaRef =
```
Comment (linzhou-db) [inline python/delta-kernel-python/src/lib.rs:90]: I assume these changes are all suggested by @nicklan

File: server/src/main/scala/io/delta/sharing/kernel/internal/DeltaSharedTableKernel.scala [hunk 2]
```diff
@@ -703,27 +703,24 @@ class DeltaSharedTableKernel(
...
-      // scalastyle:off caselocale
       if (!DeletionVectorsTableFeature.isInSet(clientReaderFeaturesSet)) {
         throw new DeltaSharingUnsupportedOperationException("Deletion Vector property disabled")
       }
-      // scalastyle:on caselocale
       val dvAbsolutePath = dvDescriptor.absolutePath(dataPath)
       val presignedDvUrl = fileSigner.sign(dvAbsolutePath)
       minUrlExpirationTimestamp = minUrlExpirationTimestamp.min(presignedDvUrl.expirationTimestamp)
       dvDescriptor = dvDescriptor.copy(pathOrInlineDv = presignedDvUrl.url, storageType = "p")
+      numRecords -= dvDescriptor.cardinality
```
Comment (linzhou-db) [inline server/src/main/scala/io/delta/sharing/kernel/internal/DeltaSharedTableKernel.scala:721]: can we add a server side test for this?

File: python/delta_sharing/tests/test_delta_sharing.py [hunk 3]
```diff
@@ -484,32 +485,229 @@ def test_get_table_protocol(profile_path: str):
...
+            ),
+            id="dv",
+        ),
+        pytest.param(
+            "share8.default.deletion_vectors_with_dvs_dv_property_on",
+            6,
+            None,
+            pd.DataFrame(
+                {
+                    "id": [3, 4, 5, 6, 7, 8],
```
Comment (linzhou-db) [inline python/delta_sharing/tests/test_delta_sharing.py:541]: How did you confirm that the returned dataframe is correct?

**Review by linzhou-db — COMMENTED** [review]
And we need

**Review by linzhou-db — APPROVED** [review]
Approving, but let's do 2 follow-ups:
- add that server side testing.
- test what will happen if we add 3.10 and 3.11

File: python/delta-kernel-python/src/lib.rs (hunk 1 above)
Comment (pranavsuku-db) [reply python/delta-kernel-python/src/lib.rs:90]: yup

File: python/delta_sharing/tests/test_delta_sharing.py (hunk 3 above)
Comment (pranavsuku-db) [reply python/delta_sharing/tests/test_delta_sharing.py:541]: i printed it out on dogfood

File: server/src/main/scala/io/delta/sharing/kernel/internal/DeltaSharedTableKernel.scala (hunk 2 above)
Comment (pranavsuku-db) [reply server/src/main/scala/io/delta/sharing/kernel/internal/DeltaSharedTableKernel.scala:721]: yup will add

File: .github/workflows/build-and-test.yml [hunk 4]
```diff
@@ -32,9 +32,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9]
```
[the remark this answers is not in the document]
Comment (pranavsuku-db) [reply .github/workflows/build-and-test.yml]: done

**Pull request merged**